indico/modules/events/sessions/controllers/management/sessions.py

```diff
@@ -0,0 +1,114 @@
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from flask import request
+
+from indico.core.db.sqlalchemy.colors import ColorTuple
+from indico.modules.events.sessions.controllers.management import RHManageSessionsBase, RHManageSessionBase
+from indico.modules.events.sessions.forms import CreateSessionForm
+from indico.modules.events.sessions.models.sessions import Session
+from indico.modules.events.sessions.operations import create_session, update_session, delete_sessions
+from indico.modules.events.sessions.util import get_colors, get_active_sessions
+from indico.modules.events.sessions.views import WPManageSessions
+from indico.web.flask.templating import get_template_module
+from indico.web.forms.base import FormDefaults
+from indico.web.util import jsonify_data, jsonify_template
+from MaKaC.webinterface.rh.base import RH
+
+
+def _render_sessions_list(event):
+    tpl = get_template_module('events/sessions/management/_sessions_list.html')
+    return tpl.render_sessions_list(event, get_active_sessions(event), get_colors())
+
+
+class RHSessionsList(RHManageSessionsBase):
+    """Display list of all sessions within the event"""
+
+    def _process(self):
+        return WPManageSessions.render_template('management/sessions_list.html', self.event_new.as_legacy,
+                                                event=self.event_new, sessions=get_active_sessions(self.event_new),
+                                                default_colors=get_colors())
+
+
+class RHAddSession(RHManageSessionsBase):
+    """Add a session to the event"""
+
+    def _process(self):
+        form = CreateSessionForm()
+        if form.validate_on_submit():
+            create_session(self.event_new, form.data)
+            return jsonify_data(sessions_list=_render_sessions_list(self.event_new))
+        return jsonify_template('events/sessions/management/add_session.html', form=form)
+
+
+class RHModifySession(RHManageSessionBase):
+    """Modify a session"""
+
+    def _process(self):
+        form = CreateSessionForm(obj=self._get_form_defaults())
+        if form.validate_on_submit():
+            update_session(self.session, form.data)
+            return jsonify_data(sessions_list=_render_sessions_list(self.event_new))
+        return jsonify_template('events/sessions/management/add_session.html', form=form)
+
+    def _get_form_defaults(self):
+        return FormDefaults(self.session, colors={'background': self.session.background_color,
+                                                  'text': self.session.text_color})
+
+
+class RHDeleteSession(RHManageSessionBase):
+    """Remove a single session"""
+
+    def _process(self):
+        return RH._process(self)
+
+    def _process_DELETE(self):
+        delete_sessions([self.session])
```

Comment:
I think it should be `delete_session` - and when you delete more than one, you call it multiple times